lib/test.js

```diff
@@ -135,27 +135,38 @@ Test.prototype.run = function () {
 	ret = observableToPromise(ret);
 
 	if (isPromise(ret)) {
-		ret
-			.then(function () {
-				if (!self.planCount || self.planCount === self.assertCount) {
-					self.exit();
-				}
-			})
-			.catch(function (err) {
-				self._setAssertError(new assert.AssertionError({
-					actual: err,
-					message: 'Promise rejected → ' + err,
-					operator: 'promise'
-				}));
-
+		if (!this.metadata.async) {
+			ret = ret.then(function () {
 				self.exit();
 			});
+		}
+
+		ret.catch(function (err) {
```

Comment:
Yes, but I see no reason you wouldn't rather use 'test()' with an async function in that case. Or even 'test' with a normal function and return a promise. I want to enforce best practices whenever possible.

Unless I'm missing something?